### DISCUSSION TITLE
객체를 복사할 때 발생할 수 있는 문제점은 무엇인가요?

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#61] [원시 값과 객체의 비교]
- 객체를 복사할 때 발생할 수 있는 문제점은 무엇인가요?

**Comment by pipisebastian** [comment]
> 객체를 복사할 때 발생할 수 있는 문제점은 무엇인가요?

- 자바스크립트에서 객체를 복사하는 방법에는 얕은 복사, 깊은 복사가 있는데, 
- 만약 얕은 복사의 경우, 참조값을 복사하기 때문에, 여러개의 식별자가 하나의 메모리주소를 공유하게 됩니다. 이는 복사된 객체를 수정하면, 원본 객체까지 변경되는 이슈가 존재합니다.

**Comment by publdaze** [comment]
> - 객체를 복사할 때 발생할 수 있는 문제점은 무엇인가요?
- 얕은 복사의 경우 참조값을 복사하기 때문에 같은 객체에 여러 변수가 접근이 가능하여 의도치 않은 데이터 훼손이 발생할 수 있음
